File: languages/dart/institution.dart
class Institution extends BaseModel {
  // Foundation
  final String premise;
  final int foundDate;
  final int endDate;
  final String parentInstitution; //ss institution
  // Claim
  final String territories; //ms territory
  final String objects; //ms object
  final String creatures; //ms creature
  final String legal; //ms law
  // World
  final String situation;
  final String cooperates; //ms institution
  final String competition; //ms institution
  final String constructs; //ms construct
  final String phenomena; //ms phenomenon
}

File: conversions/csharp_unity_json_properties/Institution.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Institution : Element
{
    [JsonProperty("premise"), TextAttribute("")]
    public string premise;
    [JsonProperty("found_date"), Integer(0)]
    public int foundDate;
    [JsonProperty("end_date"), Integer(0)]
    public int endDate;
    [JsonProperty("parent_institution"), ReferenceAttribute(typeof(Institution))]
    public string parentInstitution;
    [JsonProperty("territories"), ReferenceAttribute(typeof(Territory), true)]
    public string territories;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("legal"), ReferenceAttribute(typeof(Law), true)]
    public string legal;
    [JsonProperty("situation"), TextAttribute("")]
    public string situation;
    [JsonProperty("cooperates"), ReferenceAttribute(typeof(Institution), true)]
    public string cooperates;
    [JsonProperty("competition"), ReferenceAttribute(typeof(Institution), true)]
    public string competition;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
}
